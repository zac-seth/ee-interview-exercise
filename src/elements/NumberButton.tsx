import { ReactElement } from 'react';

import { Button } from './Button.style';
import { BasicButtonProps } from './shared';

export type NumberButtonProps = BasicButtonProps<number>;

export const NumberButton = <TValue extends unknown>({
  label,
  onClick,
  value
}: BasicButtonProps<TValue>): ReactElement => (
  <Button onClick={() => onClick(value)}>{label}</Button>
);
