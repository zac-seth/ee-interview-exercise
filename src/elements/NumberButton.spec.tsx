import { fireEvent, render, screen } from '@testing-library/react';

import { NumberButton } from './NumberButton';

describe(`NumberButton`, () => {
  it(`Should render a number button`, () => {
    const handleClick = () => {};
    render(<NumberButton label='0' onClick={handleClick} value={0} />);

    expect(screen.getByText(`0`)).toMatchSnapshot();
  });

  it(`Should invoke the handler when clicked`, () => {
    const handleClick = jest.fn();
    render(<NumberButton label='0' onClick={handleClick} value={0} />);

    fireEvent.click(screen.getByText(`0`));

    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(handleClick).toHaveBeenCalledWith(0);
  })
});
