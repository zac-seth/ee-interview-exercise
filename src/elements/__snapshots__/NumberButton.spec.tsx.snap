// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberButton Should render a number button 1`] = `
.c0 {
  font-family: 'Lato',sans-serif;
  font-weight: 400;
  color: #666;
  text-shadow: 0 0 5px #1A94D2,0 0 15px #228DFF;
  font-size: 4rem;
  line-height: 4rem;
  text-align: center;
  padding: 1rem;
  margin: 0;
  width: 8rem;
  height: 8rem;
  background-color: inherit;
  color: #1a94d2;
  border: 0;
  cursor: pointer;
}

.c0:hover {
  background-color: #1a1a1a;
}

<button
  class="c0"
>
  0
</button>
`;
